python/tk_maya/panel_util.py

```diff
@@ -58,10 +77,18 @@ def _on_parent_closed_callback(widget_id):
     """
     widget = _find_widget(widget_id)
     if widget:
-        widget.close()
-        # delete later since we are inside a slot
-        widget.deleteLater()
-    
+        # Use the proper close logic according to the Maya version.
+        if mel.eval("getApplicationVersionAsFloat()") < 2017:
+            # Close and delete the Shotgun app panel widget.
+            # It needs to be deleted later since we are inside a slot.
+            widget.close()
+            widget.deleteLater()
+        else:  # Maya 2017 and later
+            # Reparent the Shotgun app panel widget under Maya main window for later use.
+            ptr = OpenMayaUI.MQtUtil.mainWindow()
+            main_window = shiboken.wrapInstance(long(ptr), QtGui.QMainWindow)
+            widget.setParent(main_window)
```

Comment:
Ok, understood! Not a big deal if it's complicated! Sorry, didn't realise from reading the code! :)